### DISCUSSION TITLE
Assets are sent to an offline node

### DISCUSSION THREAD

**Discussion by SevenLiquor** [lightninglabs/taproot-assets#944] [Q&A] [answered]
Hello,
I'm having a little trouble with `Tapd`, hopefully someone can help me out!thx！
### question
Last Friday, I paid 100 amount of assets to an address. But to my surprise, before I made the payment, the node receiving the asset unexpectedly shut down and never came back online.Until now, I couldn't use this asset. Even though the bitcoin transactions on the chain have been confirmed for a long time.
When I use the command `tapcli assets list`,  I can't get the information about the assets that this transaction should retrieve for me.
Information about assets that have not been updated can be obtained only from the` tapcli assets balance` command 。

I want to know.The node that received the asset is not online, how do I retrieve it?

### Another question.
There's another question that I think may be related to this one。
One of my nodes received two transactions for 1000 amounts.They belong to the same asset.Then I paid 1100 amount to an address。
At this point, I should be left with about 900 dollars in assets.I can find it in the `tapcli assets list`.But if I use `tapcli assets balance` I get a 1900 amount of asset information.I think it was one of the transactions that was never confirmed by `tapd`, even though the transaction had been confirmed on the chain for a long time (more than 2 days)

Same question as the first one, I want to know how I can make this asset normal？

 I use：
litd：0.12.5
lnd：0.17.5
tapd: 0.3.3
network:regtest

**Comment by guggero** [comment]
Can you please share the logs of the sending and receiving nodes? Sounds like something went wrong there.

**Reply by SevenLiquor** [reply]
I'm very sorry I wasn't able to save the log from that time. 
But I would like to ask some questions. When using the universe to pass proof of assets, does the sender need to wait for an acknowledgement signal or something else from the receiver after sending the assets?As well as at the time of proof file courier, the sender's remaining assets are unavailable?
very thx！

**Reply by guggero** [reply]
Transfers are only finalized after one confirmation, because the header of the block the transaction is confirmed in is part of the proof. That's why the proof is only complete after a confirmation and therefore will only be transferred after it's in a block.
So yes, you need to wait for at least one confirmation on the sending side before you can stop the node. And also until the proof was uploaded to the courier, which you can detect using the notification RPCs.
On the sending side the asset (e.g. the change output) is also unavailable until it has been confirmed.

**Reply by SevenLiquor** [reply]
So if the node receiving the asset suddenly goes offline before I send it. Will my asset remain unavailable until the receiver comes back online?

**Reply by guggero** [reply]
No, the proof transfer is independent from the asset change availability on the sender. I was just saying that both only happen after things confirm on chain.

**Reply by SevenLiquor** [reply]
Thanks for the explanation. Because I sent an asset to an offline node earlier, but my remaining assets have been in an unavailable state. So I have this problem. Unfortunately I didn't keep a good log file of it, I'll take care of it next time.
Have a nice day!

**Reply by fanshome1001** [reply]
> Transfers are only finalized after one confirmation, because the header of the block the transaction is confirmed in is part of the proof. That's why the proof is only complete after a confirmation and therefore will only be transferred after it's in a block. So yes, you need to wait for at least one confirmation on the sending side before you can stop the node. And also until the proof was uploaded to the courier, which you can detect using the notification RPCs. On the sending side the asset (e.g. the change output) is also unavailable until it has been confirmed.

If the proof document can only be constructed after the transaction is confirmed, then in the construction of an atomic transaction of bitcoin and assets, once the on-chain confirmation is obtained, then the seller of the asset has obtained the Bitcoin, while the buyer of the asset may not have the proof document, and the atomic transaction will not be established.

**Reply by guggero** [reply, marked as answer]
A transition proof is only fully finalized (anchored to the proof of work of the Bitcoin blockchain) after it has been confirmed. But such a proof consists of multiple parts with attestations/signatures at multiple levels (signature/witness at the asset level, inclusion/exclusion proofs using MS-SMT roots, signature/witness at the BTC level), all of which are created at the transaction creation time. Only the "thermodynamic seal of immutability" (e.g. timestamp) of the Proof of Work is missing. So an atomic swap with assurances for both the seller and buyer is still possible. And in such an interactive (e.g. both parties communicate to arrive at the transition proof) scenario, the parties also don't have to wait for the confirmation to receive a proof file. They both contribute to creating the transaction and the (partial) proofs, then they independently can finish the block header/PoW part once the confirmation comes in.

So it's very similar to how a normal collaborative BTC transaction (e.g. atomic swap, coin join, pay join and so on) happens: Both parties communicate, exchange signatures and so on, and then arrive at a finalized and fully signed BTC transaction. Once that's published to the mempools of the network, no communication is necessary between the participants anymore. But they all have to wait for the transaction to be confirmed to be certain it is fully finalized.
